### DISCUSSION TITLE
z-index를 SCSS 변수 관리하듯 프로젝트 차원에서 관리하자

### DISCUSSION THREAD

**Discussion by JiinSeok** [team-final/albaform#50] [General]
z-index를 SCSS 변수 관리하듯 프로젝트 차원에서 관리하자

**Comment by JiinSeok** [comment]
1. 전역 O 루트 레이아웃

모달 - 알럿
모달 - 다이얼로그

GNB

헤더, 푸터

2. 전역 X

바디
- 플로팅버튼
- 드롭다운
- 콘텐트

**Reply by JiinSeok** [reply]
어쨌든 정하니까 좋네요... 고민할 시간 아낄 것 같다.

**Reply by JiinSeok** [reply]
// 설정파일

플로팅버튼 z-index를 변수로 100으로 정해놓으면
컴포넌트에서는, 100 말고 변수를 입력하면 된다.

**Reply by JiinSeok** [reply]
부모 요소 안의 범위로 한정되는 z-index 특성상 이렇게만 관리해서는 무의할 것 같다.

**Reply by JuhyeokC** [reply]
body 의 직계자식 너머 손자들의 값은 부모가 가진 값을 못넘는데 값의 관리가 필요할까??

어떤 목적일까?

body 직계자식로 관리되는 컴포넌트정도만 z-index 를 부여해놓으면 되지 않을까?
추가로 생성되는 컴포넌트들은 생성할 때 값을 고민하는 것으로 충분하지 않을까?

예시로 Modal: 32, GNB: 16, Container: 8 을 정해두고

컨테이너의 컨텐츠 보다는 상위지만 GNB 를 가리지 않아야하는 건 8초과 16미만의 값으로 지정한다.
특수목적(애니메이션 등)으로 컨테이너 아래로 숨겨야하는 것은 8미만의 값으로 지정하기.

미팅 때 말씀드린 것처럼
'SCSS 변수 관리하듯' 프로젝트 차원에서 개념정도만 정의해두면 될 것 같습니다
